### PR TITLE
Handle multiple tables in DatabaseManager (creation and deletion)

### PR DESCRIPTION
**Pull request by nikitag-ms**
* [ ] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [x] Did you add unit tests?
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?
* [x] Did you check UI tests on the sample app? They are not executed on CI.

## Description

Add `create` and `drop` table operations to `DatabaseManager`

### CODE REVIEW

File: sdk/appcenter/src/androidTest/java/com/microsoft/appcenter/utils/storage/DatabaseManagerAndroidTest.java [hunk 1]
```diff
@@ -489,4 +490,40 @@ public void replaceByMultipleColumns() {
...
+        assertFalse(isTableExists(databaseManager.getDatabase(), firstTable));
+        assertTrue(isTableExists(databaseManager.getDatabase(), secondTable));
+
+        databaseManager.dropTable(secondTable);
+        assertFalse(isTableExists(databaseManager.getDatabase(), firstTable));
+        assertFalse(isTableExists(databaseManager.getDatabase(), secondTable));
+    }
+
+    private boolean isTableExists(SQLiteDatabase db, String tableName) {
+        Cursor cursor = db.rawQuery("select DISTINCT tbl_name from sqlite_master where tbl_name = '" + tableName + "'", null);
```
Comment (guperrot) [inline sdk/appcenter/src/androidTest/java/com/microsoft/appcenter/utils/storage/DatabaseManagerAndroidTest.java]: Would be better to just put a value and read it. With different schemas.

File: sdk/appcenter/src/main/java/com/microsoft/appcenter/utils/storage/DatabaseManager.java [hunk 2]
```diff
@@ -514,6 +512,28 @@ void setSQLiteOpenHelper(@NonNull SQLiteOpenHelper helper) {
         mSQLiteOpenHelper = helper;
     }
 
+    private void createTable(SQLiteDatabase db, String table) {
+        /* Generate a schema from specimen. */
```
Comment (guperrot) [inline sdk/appcenter/src/main/java/com/microsoft/appcenter/utils/storage/DatabaseManager.java]:
```suggestion
        
        /* Generate a schema from specimen. */
```

File: sdk/appcenter/src/main/java/com/microsoft/appcenter/utils/storage/DatabaseManager.java [hunk 3]
```diff
@@ -170,6 +151,23 @@ private static ContentValues buildValues(Cursor cursor, ContentValues schema) {
         return values;
     }
 
+    /**
+     * Creates a new table in the database.
+     * @param table name.
+     */
+    public void createTable(String table) {
```
Comment (guperrot) [inline sdk/appcenter/src/main/java/com/microsoft/appcenter/utils/storage/DatabaseManager.java]: Use schema as parameter here.

Comment (guperrot) [inline sdk/appcenter/src/main/java/com/microsoft/appcenter/utils/storage/DatabaseManager.java]:
```suggestion
    public void createTable(@NonNull String table) {
```

File: sdk/appcenter/src/main/java/com/microsoft/appcenter/utils/storage/DatabaseManager.java [hunk 4]
```diff
@@ -170,6 +151,23 @@ private static ContentValues buildValues(Cursor cursor, ContentValues schema) {
...
+        createTable(getDatabase(), table);
+    }
+
+    /**
+     * Deletes a table in the database.
+     * @param table name.
+     */
+    public void dropTable(String table) {
```
Comment (guperrot) [inline sdk/appcenter/src/main/java/com/microsoft/appcenter/utils/storage/DatabaseManager.java]:
```suggestion
    public void dropTable(@NonNull String table) {
```

File: sdk/appcenter/src/main/java/com/microsoft/appcenter/utils/storage/DatabaseManager.java [hunk 5]
```diff
@@ -170,6 +151,23 @@ private static ContentValues buildValues(Cursor cursor, ContentValues schema) {
...
+        SQLiteDatabase db = getDatabase();
+        db.execSQL(String.format("DROP TABLE %s", table));
```
Comment (guperrot) [inline sdk/appcenter/src/main/java/com/microsoft/appcenter/utils/storage/DatabaseManager.java]:
```suggestion
        db.execSQL(String.format("DROP TABLE `%s`", table));
```

File: sdk/appcenter/src/androidTest/java/com/microsoft/appcenter/utils/storage/DatabaseManagerAndroidTest.java [hunk 6]
```diff
@@ -489,4 +490,40 @@ public void replaceByMultipleColumns() {
         }
         verify(listener).onCreate(any(SQLiteDatabase.class));
     }
+
+    @Test
+    public void testMultipleTables() {
+        final String firstTable = "firstTable";
```
Comment (guperrot) [inline sdk/appcenter/src/androidTest/java/com/microsoft/appcenter/utils/storage/DatabaseManagerAndroidTest.java]:
```suggestion
       String firstTable = "firstTable";
```

File: sdk/appcenter/src/androidTest/java/com/microsoft/appcenter/utils/storage/DatabaseManagerAndroidTest.java [hunk 7]
```diff
@@ -489,4 +490,40 @@ public void replaceByMultipleColumns() {
...
+        final String secondTable = "secondTable";
```
Comment (guperrot) [inline sdk/appcenter/src/androidTest/java/com/microsoft/appcenter/utils/storage/DatabaseManagerAndroidTest.java]:
```suggestion
       String secondTable = "secondTable";
```

File: sdk/appcenter/src/androidTest/java/com/microsoft/appcenter/utils/storage/DatabaseManagerAndroidTest.java (hunk 1 above)
Comment (nikitag-ms) [reply sdk/appcenter/src/androidTest/java/com/microsoft/appcenter/utils/storage/DatabaseManagerAndroidTest.java]: Sure, I'll add write-read tests as well, but I still think it's valuable to check that there are indeed tables underneath.

Comment (nikitag-ms) [reply sdk/appcenter/src/androidTest/java/com/microsoft/appcenter/utils/storage/DatabaseManagerAndroidTest.java]: Still working on that new test. For some reason, the write to the second table doesn't actually add a record in there (though no exception and I checked in the DB browser, it's actually not there).

Comment (nikitag-ms) [reply sdk/appcenter/src/androidTest/java/com/microsoft/appcenter/utils/storage/DatabaseManagerAndroidTest.java]: Ok, that was a typo. Fixed it.

Comment (guperrot) [reply sdk/appcenter/src/androidTest/java/com/microsoft/appcenter/utils/storage/DatabaseManagerAndroidTest.java]: If put/read works then the table has to exist. We can remove this code which is basically using unsecure query that can be picked up by a code scanner.

**Pull request merged**